Dao

STOzaki opened the pull request:
Created Daos

STOzaki committed:
setup java project

STOzaki committed:
models added

STOzaki committed:
models added

STOzaki committed:
models added

kingjon3377 committed:
Loan: Make book, branch, and borrower fields final.

These fields together constitute the identity of a given loan,
and so should not be allowed to be changed.

kingjon3377 committed:
In model classes, use only identity fields in hashCode().

In most cases, this means their ID number field; in Loan,
the book, borrower, and branch. This prevents memory leaks,
as hashing based on fields that can be changed by setters
(or themselves have hashCode() methods that could change)
causes HashSets and HashMaps to lose the hashed object
in the old hash bucket.

STOzaki committed:
daos added

kingjon3377 committed:
Condense all model-object equals() methods.

The algorithm is essentially the same, but this cuts the
line count by about half, and may also add some
null-safety.

STOzaki committed:
Merge pull request #2 from kingjon3377/t/condense-equals-hash

Improve equals() and hashCode()

al-amine reviewed (approved):
approved

STOzaki committed:
removed redundancy

al-amine commented on src/com/lms/model/Branch.java:
```diff
@@ -33,35 +35,19 @@ public int getId() {
 	}
 
 	@Override
-	public boolean equals(Object obj) {
```
it would be better if you override equals() using only the Id of the object without the other fields !

STOzaki replied:
Yes, which Jonathan corrected. I updated this branch with those changes, so we should be good now.

STOzaki committed:
removed redundancy

kingjon3377 reviewed (approved):
I generally approve; there are some mostly-cosmetic changes that I will make in my fork (removing redundant modifiers, making documentation use Javadoc syntax, etc.), but none of that should hold this back.

STOzaki commented:
> I generally approve; there are some mostly-cosmetic changes that I will make in my fork (removing redundant modifiers, making documentation use Javadoc syntax, etc.), but none of that should hold this back.

Well, can you make those changes and merge them in, like what you did before?

kingjon3377 commented:
> Well, can you make those changes and merge them in, like what you did before?

I can, but unlike my earlier PR to your fork (the changes I'm expecting to make to this code won't affect anything that uses it.

STOzaki commented:
> > Well, can you make those changes and merge them in, like what you did before?
> 
> I can, but unlike my earlier PR to your fork (the changes I'm expecting to make to this code won't affect anything that uses it.

I guess that is up to you. It would be great if you can because you are amazing at commenting.

kingjon3377 committed:
Add 'throws SQLException' to signature of all DAO methods.

kingjon3377 committed:
In DAO classes, make create() methods return the created object.

STOzaki committed:
add jdbc to lib and added authentication for connection to mySQL

STOzaki committed:
Book Loans does not extend Dao

STOzaki committed:
Logger and mysql connection jars added to project

STOzaki committed:
correct labeling

kingjon3377 committed:
Rename book-loans DAO interface from BookLoans to BookLoansDao.

kingjon3377 committed:
BookLoansDao: Fix method signatures to refer to Loan identity.

get() and getAll() need to return a Loan and a List of Loans,
respectively, and update() and delete() need to take Loan
parameters, not Books.

kingjon3377 commented on .classpath:
```diff
@@ -0,0 +1,13 @@
+	<classpathentry kind="src" path="test"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.junit.JUNIT_CONTAINER/5"/>
+	<classpathentry kind="lib" path="C:/Users/salem/eclipse-workspace/NovaTechLMSSQL/lib/mysql-connector-java-5.1.47.jar"/>
```
This definitely needs to be changed from an absolute path to one relative to the project root.

STOzaki replied:
So, you mean something like lib/mysql-connector-java-5.1.47.jar?

kingjon3377 replied:
Right. You used relative paths for the other libraries you added to `.classpath`, but this one won't work on anyone else's computer as it stands.

STOzaki replied:
You are right. I changed it and pushed it.

STOzaki committed:
add toString method to all models
